acracensor/handlers/querycapture_handler.go

```diff
@@ -5,89 +5,127 @@ import (
 	"io/ioutil"
 	"encoding/json"
 	"os"
+	"github.com/cossacklabs/acra/logging"
+	log "github.com/sirupsen/logrus"
+	"time"
+	"os/signal"
+	"syscall"
 )
 
+const MaxQueriesInChannel = 10
+const TimeoutSecondsToSerialize = 1
+
 type QueryCaptureHandler struct {
 	Queries []QueryInfo
 	filePath string
-	occuredError *CaptureError
-
 	logChannel chan QueryInfo
+	signalToSerialize chan bool
 }
 
 type QueryInfo struct {
 	RawQuery string
 	IsForbidden bool
 }
 
-type CaptureError struct {
-	err error
-	query *QueryInfo
-}
-
 func NewQueryCaptureHandler(filePath string) (*QueryCaptureHandler, error) {
-	file, err := os.OpenFile(filePath, os.O_RDONLY|os.O_CREATE, 0600)
-	if err != nil {
-		return nil, err
+
+	var _, err = os.Stat(filePath)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
```

Comment:
you don't check other errors than IsNotExist. As example user hasn't permission to this file, or device is busy or anything else
```
if err != nil {
   if os.IsNotExist(err) {
   } else {
      // handle other errors
   }
}
```